src/server/v2.0/handler/base.go

```diff
@@ -121,26 +121,44 @@ func (b *BaseAPI) BuildQuery(ctx context.Context, query *string, pageNumber, pag
 
 // Links return Links based on the provided pagination information
 func (b *BaseAPI) Links(ctx context.Context, u *url.URL, total, pageNumber, pageSize int64) internal.Links {
-	url := *u
+	ul := *u
+	// try to unescape the repository name which contains escaped slashes
+	if escapedPath, err := url.PathUnescape(ul.Path); err == nil {
+		ul.Path = escapedPath
+	} else {
+		log.Errorf("failed to unescape the path %s: %v", ul.Path, err)
+	}
 	var links internal.Links
 	// prev
 	if pageNumber > 1 && (pageNumber-1)*pageSize < total {
-		q := url.Query()
+		q := ul.Query()
 		q.Set("page", strconv.FormatInt(pageNumber-1, 10))
-		url.RawQuery = q.Encode()
+		ul.RawQuery = q.Encode()
+		// try to unescape the query
+		if escapedQuery, err := url.QueryUnescape(ul.RawQuery); err == nil {
```

Comment:
Why do we need to unescape query string?

Reply:
pkg/q/builder.Build() can only handle the unescaped characters

Reply:
I think the `q` is unescaped automic before pass it the `pkg/q/builder.Build`, is it right?

And we need to response the same values as they are in the query string, we only change the `page` paramater.

Reply:
In line 136, we encode the query string, the special characters used in our query(~,[],etc) are all encoded, so here decode them first